vortex-array/src/arrow/convert.rs

```diff
@@ -434,6 +435,18 @@ impl FromArrowArray<&dyn ArrowArray> for ArrayRef {
     }
 }
 
+impl FromArrowArray<RecordBatch> for ArrayRef {
+    fn from_arrow(array: RecordBatch, nullable: bool) -> Self {
+        ArrayRef::from_arrow(&arrow_array::StructArray::from(array), nullable)
+    }
+}
+
+impl FromArrowArray<&RecordBatch> for ArrayRef {
+    fn from_arrow(array: &RecordBatch, nullable: bool) -> Self {
+        ArrayRef::from_arrow(&arrow_array::StructArray::from(array.clone()), nullable)
+    }
+}
```

Comment:
We also have a `TryIntoArray` trait which whole purpose is to implement this. Can you delete & replace usages while you're doing this?